### PR TITLE
Closes #8065: Support subdomain stripping for "mobile." URLs

### LINKED ISSUE #8065: Support subdomain stripping for "mobile." URLs
Currently we only support "m." subdomain stripping.  Another generally used subdomain for mobile devices is "mobile.".  Add support for "mobile." subdomain stripping.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt [hunk 1]
```diff
@@ -146,6 +147,7 @@ class AppLinksInterceptor(
         return when {
             url == null -> return null
             url.startsWith(WWW) -> url.replaceFirst(WWW, "")
+            url.startsWith(M) -> url.replaceFirst(M, "")
```
Comment (csadilek) [inline components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:150]: @psymoon also landed code recently that checks for mobile URls. I think soon we'll want a share utility / extension function? https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt#L275

Comment (rocketsroger) [reply components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:150]: That's great.  I'll keep an eye out for the shared utility / extension function.

Comment (rocketsroger) [reply components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:150]: Like you mentioned offline that there's not a lot of duplication right now.  I'll keep this in mind if I make anymore changes in the future relating to mobile URLs.  Right now I agree this change is not introducing too much duplication.  I'll merge it.  Thanks,

**Comment by rocketsroger** [comment]
bors r=csadilek

**Pull request merged**